src/HealthChecks.Redis/RedisHealthCheck.cs

```diff
@@ -24,7 +24,16 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
         {
             if (!_connections.TryGetValue(_redisConnectionString, out var connection))
             {
-                connection = await ConnectionMultiplexer.ConnectAsync(_redisConnectionString).ConfigureAwait(false);
+                var timeoutTask = Task.Delay(Timeout.Infinite, cancellationToken);

```

Comment:
Please see https://github.com/graphql-dotnet/server/pull/985 to how to avoid leaking `Task.Delay` instances.

Reply:
Thanks for the feedback,
I learned something new today 😄 

I also realized that since we didn't actually cancel the `ConnectionMultiplexer`, it's still trying to constantly reconnect in the background, even though the healthcheck has failed. So every time the healthcheck times out and we hit the healthcheck again, we will keep adding to the `ConnectionMultiplexer`'s attempts to reconnect. I dont know if the `ConnectionMultiplexer` is smart enough to just continue reconnecting again using the last attempt or will it start a whole new set of re-attempts. This might cause problems if people just keep hitting the healthcheck and don't actually restart their applications. 

Ideally if the health check is failing, something bad is happening anyway... so this should be ok?
And once the connection is established, things should resolve themselves.
It's sort of a corner case

Reply:
I don't know internals of `ConnectionMultiplexer`.